core/chains/evm/client/simulated_backend_client.go

```diff
@@ -466,6 +477,8 @@ func (c *SimulatedBackendClient) BatchCallContext(ctx context.Context, b []rpc.B
 			b[i].Error = c.ethGetHeaderByNumber(ctx, b[i].Result, b[i].Args...)
 		case "eth_estimateGas":
 			b[i].Error = c.ethEstimateGas(ctx, b[i].Result, b[i].Args...)
+		case "eth_getLogs":
+			b[i].Error = c.ethGetLogs(ctx, b[i].Result, b[i].Args...)
```

Comment:
Out of scope, but it looks like we could capture the methods as `func(context.Context, interface{}, ...interface{}) error`s here to simplify this section:
```go
var method func(context.Context, interface{}, ...interface{}) error
switch elem.Method {
case "eth_getTransactionReceipt":
	method = c.ethGetTransactionReceipt
case "eth_getBlockByNumber":
	method = c.ethGetBlockByNumber
case "eth_call":
	method = c.ethCall
case "eth_getHeaderByNumber":
	method = c.ethGetHeaderByNumber
case "eth_estimateGas":
	method = c.ethEstimateGas
case "eth_getLogs":
	method = c.ethGetLogs
default:
	return fmt.Errorf("SimulatedBackendClient got unsupported method %s", elem.Method)
}
b[i].Error = method(ctx, b[i].Result, b[i].Args...)
```

Reply:
Might as well, while we're updating this file. Just made that change, and replaced a `panic("can never reach here")` message I found (in a reachable code path!) with a more useful error message